demos/background_subtraction_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,184 @@
+        if (!tmp.data) {
+            throw std::runtime_error("Couldn't grab first frame");
+        }
+        cv::Size frame_size = cv::Size{tmp.cols, tmp.rows};
+        cap = openImagesCapture(FLAGS_i, FLAGS_loop, 0,
+            std::numeric_limits<size_t>::max(), stringToSize(FLAGS_res));
+
+        auto pipeline = comp.compileStreaming(cv::compile_args(custom::kernels(),
+                                                               cv::gapi::networks(net)));
+
+        /** Output containers for results **/
+        cv::Mat output;
+
+        /** ---------------- The execution part ---------------- **/
+        auto pipeline_inputs = cv::gin(cv::gapi::wip::make_src<custom::CommonCapSrc>(cap));
+        if (!FLAGS_blur_bgr && FLAGS_target_bgr.empty()) {
+            cv::Scalar default_color(155, 255, 120);
+            pipeline_inputs += cv::gin(cv::Mat(frame_size, CV_8UC3, default_color));
+        } else if (!FLAGS_target_bgr.empty()) {
+            std::shared_ptr<ImagesCapture> target_bgr_cap = openImagesCapture(FLAGS_target_bgr, true, 0,
+                std::numeric_limits<size_t>::max());
+            pipeline_inputs += cv::gin(cv::gapi::wip::make_src<custom::CommonCapSrc>(target_bgr_cap));
+        }
+
+        pipeline.setSource(std::move(pipeline_inputs));
+        std::string windowName = "Background subtraction demo G-API";
+        int delay = 1;
+
+        cv::Size graphSize{static_cast<int>(frame_size.width / 4), 60};
+        Presenter presenter(FLAGS_u, frame_size.height - graphSize.height - 10, graphSize);
+
+        /** Save output result **/
+        cv::VideoWriter videoWriter;
+        if (!FLAGS_o.empty() && !videoWriter.open(FLAGS_o, cv::VideoWriter::fourcc('M', 'J', 'P', 'G'),
+                                                  cap->fps(), frame_size)) {
+            throw std::runtime_error("Can't open video writer");
+        }
+
+        bool isStart = true;
+        uint64_t curr_frame_num = 0;
+        const auto startTime = std::chrono::steady_clock::now();
+        pipeline.start();
+
+        while(pipeline.pull(cv::gout(output))) {
+            ++curr_frame_num;
+            presenter.drawGraphs(output);
+            if (isStart) {
+                metrics.update(startTime, output, { 10, 22 }, cv::FONT_HERSHEY_COMPLEX,
+                    0.65, { 200, 10, 10 }, 2, PerformanceMetrics::MetricTypes::FPS);
+                isStart = false;
+            }
+            else {
+                metrics.update({}, output, { 10, 22 }, cv::FONT_HERSHEY_COMPLEX,
+                    0.65, { 200, 10, 10 }, 2, PerformanceMetrics::MetricTypes::FPS);
+            }
+
+            if (videoWriter.isOpened() &&
+                (FLAGS_limit <= 0 || curr_frame_num < FLAGS_limit)) {
+                videoWriter.write(output);
+            }
+
+            if (!FLAGS_no_show) {
+                cv::imshow(windowName, output);
+                int key = cv::waitKey(delay);
+                presenter.handleKey(key);
```

Comment:
There is should be the verification that the user presses `Esc` or `Q` to quit the application.

Reply:
Done